src/wp-includes/theme-templates.php

```diff
@@ -1,5 +1,33 @@
 <?php
 
+/**
+ * Sets a custom slug when creating auto-draft template parts.
+ * This is only needed for auto-drafts created by the regular WP editor.
+ * If this page is to be removed, this won't be necessary.
+ *
+ * @param int $post_id Post ID.
+ */
+function wp_set_unique_slug_on_create_template_part( $post_id ) {
+	$post = get_post( $post_id );
+	if ( 'auto-draft' !== $post->post_status ) {
+		return;
+	}
+
+	if ( ! $post->post_name ) {
+		wp_update_post(
+			array(
+				'ID'        => $post_id,
+				'post_name' => 'custom_slug_' . uniqid(),
+			)
+		);
+	}
+
+	$terms = get_the_terms( $post_id, 'wp_theme' );
+	if ( ! $terms || ! count( $terms ) ) {
```

Comment:
Fatal error in PHP 8.x
Warning in < PHP 8.0

How? if `get_the_terms()` returns `WP_Error` or `false`.

Why? The PHP native `count` requires a `Countable` or `array` type. See [PHP manual](https://www.php.net/manual/en/function.count.php) and [in action here](https://3v4l.org/c7nB7#v8.1rc3).

To guard against these problems, change the first conditional to check if what's returned is not an array:
```suggestion
	if ( ! is_array( $terms ) || ! count( $terms ) ) {
```